src/Ulid.php

```diff
@@ -87,4 +92,58 @@ public function __toString()
     {
         return $this->string();
     }
+
+    //"0123456789ABCDEFGHJKMNPQRSTVWXYZ"
+    protected static function crockford()
+    {
+        static $base32;
+
+        if (!isset($base32)) {
+            $base32 = new \Tuupola\Base32([
+                "characters" => \Tuupola\Base32::CROCKFORD,
+                "padding" => false
+            ]);
+        }
+        return $base32;
+    }
+
+    public static function sanitizeString($value)
+    {
+        $value = str_replace(
+            ['O', 'L', 'I'],
+            ['0', '1', '1'],
+            strtoupper($value)
+        );
+
+        $value = str_pad($value, 26, "0", STR_PAD_LEFT);
+
+        return static::isStringAnUlid($value) ? $value : false;
+    }
+
+    public static function fromString($value)
+    {
+
+        $timestamp = substr($value, 0, 10);
+        $payload = substr($value, 10);
+
+        return new static($timestamp, $payload, true);
+    }
+
+    public static function isStringAnUlid($value)
```

Comment:
Not sure if this is needed. Instead `UlidFactory::fromString()` should throw if given an invalid string.